docs/cli/nn.rst

```diff
@@ -0,0 +1,93 @@
+Training, Prediction, and Hyperparameter Search for Neural Networks
+===================================================================
+
+Training
+^^^^^^^^
+
+In the training procedure, you can build a model from scratch or start from some pre-obtained information.
+
+.. code-block:: bash
+
+    python3 main.py --config CONFIG_PATH \
+                    [--load_checkpoint CHECKPOINT_PATH] \
+                    [--embed_file EMBED_NAME_OR_EMBED_PATH] \
+                    [--vocab_file VOCAB_CSV_PATH]
+
+- **config**: configure parameters in a yaml file. See ``python3 main.py --help``.
+
+If a model was trained before by this package, the training procedure can start with it.
+
+- **load_checkpoint**: specify the path to a pre-trained model.
+
+To use your own word embeddings or vocabulary set, specify the following parameters:
+
+- **embed_file**: choose one of the pretrained embeddings defined in `torchtext <https://pytorch.org/text/0.9.0/vocab.html#torchtext.vocab.Vocab.load_vectors>`_ or specify the path to your word embeddings with each line containing a word followed by its vectors. Example:
+
+.. code-block::
+
+    the 0.04656 0.21318 -0.0074364 ...
+    a -0.29712 0.094049 -0.096662 ...
+    an -0.3206 0.43316 -0.086867 ...
+
+- **vocab_file**: set the file path to a predefined vocabulary set that contains lines of words.
+
+.. code-block::
+
+    the
+    a
+    an
+
+For the validation process in the training procedure, you can evaluate the model with a set of evaluation metrics. Set ``monitor_metrics`` to define what you want to print on the screen. The argument ``val_metric`` is the metric for picking the best model. Example:
```

Comment:
The model occurred at the epoch with the best validation metric is returned after training. If  users did not specify a validation set in the configuration file or via --??, we ?? (mention the 80-20 split?)